app/common/gristUrls.ts

```diff
@@ -181,12 +181,20 @@ export interface OrgUrlInfo {
   orgInPath?: string;     // If /o/{orgInPath} should be used to access the requested org.
 }
 
-function isDocInternalUrl(host: string) {
-  if (!process.env.APP_DOC_INTERNAL_URL) { return false; }
-  const internalUrl = new URL('/', process.env.APP_DOC_INTERNAL_URL);
+function isInternalUrl(host: string, envValue?: string) {
+  if (!envValue) { return false; }
+  const internalUrl = new URL('/', envValue);
   return internalUrl.host === host;
 }
```

Comment:
> As an aside, I'm slightly puzzled this works at all since APP_DOC_INTERNAL_URL is only defined inside a doc worker, and can contain only the URL for this worker, whose host isn't guaranteed to match any other worker's host.

Thanks, this sentence led me to change the functions, especially the naming to make it clearer: https://github.com/gristlabs/grist-core/pull/915/commits/52f559984069484f4eb332bb6072d96934d11c9e

The idea is that is should only be tested to check whether it matches its *own* internal URL, which I hope it is clearer now.